src/core/api.c

```diff
@@ -658,6 +658,15 @@ MsQuicStreamOpen(
         goto Error;
     }
 
+    if (!!(Flags & QUIC_STREAM_OPEN_FLAG_EXTERNAL_BUFFERS) &&
+        Connection->Settings.StreamMultiReceiveEnabled) {
```

Comment:
What if they want to use multi-receive for streams that don't use external buffers? I think we can't block that.

Reply:
You would allow an "app buffer" stream when multi-receive is enabled?
The problem is that this stream would not support the multi-receive behavior (at least for now, we could make it happen), which could be confusing for the app.

And if app takes dependency on the fact that "app buffer" stream ignore multi-receive, then we can't add support without an API break or new flags.